test-loop-tests/src/tests/gas_keys.rs

```diff
@@ -0,0 +1,281 @@
+use itertools::Itertools;
```

Comment:
please move this test to a separate PR

Reply:
I'm happy to do this as the last step in the PR and for a separate proper review of the test.
If I move it to another branch now I will have to re-apply all the refactoring to it separately which is not productive.
Additionally, it shows that the changes work at least to some extent.